src/req_rep.rs

```diff
@@ -119,6 +151,7 @@ impl MultiPeer for ReqSocketBackend {
                 _io_close_handle: stop_handle,
```

Comment:
Hmm, why do we need to synchronise access to `in_queue_receiver`? Isn't the receiving end of this channel only used in `ReqSocket::recv()`?